### PR TITLE
fix: parsing of numeric env vars

### PR DESCRIPTION
**Pull request by bguiz**
**Description**:

- added `parseNumericEnvVar` to `packages/relay/src/formatters.ts` which handles edge cases like empty strings, undefined, and other non-numeric values
- modified `Overrideable options used when initializing.` section of `packages/relay/src/lib/eth.ts` to make use of `parseNumericEnvVar`
- added units tests for `parseNumericEnvVar`

**Related issue(s)**:

- Fixes [#1409 fix: caching of block number](https://github.com/hashgraph/hedera-json-rpc-relay/issues/1409)

**Notes for reviewer**:

- See #1409

**Checklist**

- [ ] Documented (Code comments, README, etc.) --> N/A
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
 Implementation of an Ethereum JSON RPC APIs for Hedera Hashgraph. Utilises both Hedera Consensus Nodes and Mirror nodes
 to support RPC queries as defined in
-the [JSON RPC Specification](https://playground.open-rpc.org/?schemaUrl=https://raw.githubusercontent.com/hashgraph/hedera-json-rpc-relay/main/docs/openrpc.json&uiSchema%5BappBar%5D%5Bui:splitView%5D=false&uiSchema%5BappBar%5D%5Bui:input%5D=false&uiSchema%5BappBar%5D%5Bui:examplesDropdown%5D=false)
+the [JSON RPC Specification](https://playground.open-rpc.org/?schemaUrl=https://raw.githubusercontent.com/hashgraph/hedera-json-rpc-relay/release/0.26/docs/openrpc.json&uiSchema%5BappBar%5D%5Bui:splitView%5D=false&uiSchema%5BappBar%5D%5Bui:input%5D=false&uiSchema%5BappBar%5D%5Bui:examplesDropdown%5D=false)
```
Comment (Nana-EC) [inline README.md:19]: Please revert this change to main. You potentially created your feature branch off of `release/0.26`

File: packages/relay/src/formatters.ts [hunk 2]
```diff
@@ -65,4 +65,25 @@ const formatTransactionId = (transactionId: string): string | null => {
...
+/**
+ * Reads a value loaded up from the `.env` file, and converts it to a number.
+ * If it is not set in `.env` or set as an empty string or other non-numeric
+ * value, it uses the default value specified in constants.
+ * @param envVarName The name of the env var to read in from the `.env` file
+ * @param constantName The name of the constant to use as a fallback when the
+ *   specified env var is invalid
+ * @throws An error if both the env var and constant are invalid
+ */
+const parseNumericEnvVar = (envVarName: string, constantName: string): number => {
```
Comment (Nana-EC) [inline packages/relay/src/formatters.ts:77]:
```suggestion
const parseNumericEnvVar = (envVarName: string, defaultConstantKey: string): number => {
```
or
```suggestion
const parseNumericEnvVar = (envVarName: string, fallbackConstantKey: string): number => {
```

**Review by Nana-EC — CHANGES_REQUESTED** [review]
Started reviewing but there's a lot of changes unrelated, can  you rebase off of `main` branch and repush

File: packages/relay/tests/lib/formatters.spec.ts [hunk 3]
```diff
@@ -76,5 +77,73 @@ describe('Formatters', () => {
...
             expect(formatTransactionId(invalidInputTimestamp)).to.eq(null);
         })
-    })
-});
\ No newline at end of file
+    });
+
+    describe('parseNumericEnvVar', () => {
+        before(() => {
+            // process.env.TEST_ONLY_ENV_VAR_UNDEFINED = undefined;
```
Comment (Nana-EC) [inline packages/relay/tests/lib/formatters.spec.ts:84]: nit:  
```suggestion
```

File: packages/relay/tests/lib/formatters.spec.ts [hunk 4]
```diff
@@ -76,5 +77,73 @@ describe('Formatters', () => {
...
+            process.env.TEST_ONLY_ENV_VAR_EMPTYSTRING = '';
+            process.env.TEST_ONLY_ENV_VAR_NONNUMERICSTRING = 'foobar';
+            process.env.TEST_ONLY_ENV_VAR_NUMERICSTRING = '12345';
+        });
+
+        after(() => {
+            // process.env.TEST_ONLY_ENV_VAR_UNDEFINED = undefined;
```
Comment (Nana-EC) [inline packages/relay/tests/lib/formatters.spec.ts:91]: nit
```suggestion
```

**Review by Nana-EC — APPROVED** [review]
LG
Left 2 nits to remove comments.
Also assuming this ensure the relay doesn't start with bad int env values, which is preferred that what you encountered.
Thanks for the contribution

**Comment by bguiz** [comment]
@Nana-EC addressed the nitpick comments with new commit just pushed.

> LG Left 2 nits to remove comments.

Yes, for this specific subset on env values - the rest (probably) should be addressed separately.

> Also assuming this ensure the relay doesn't start with bad int env values, which is preferred that what you encountered.

**Review by Nana-EC — APPROVED** [review]
LGTM

**Review by AlfredoG87 — APPROVED** [review]
LGTM 👍
